daprdocs/content/en/developing-applications/building-blocks/state-management/query-state-store-api.md

```diff
@@ -0,0 +1,265 @@
+| `IN`        | key:[]value| key == value[0] OR key == value[1] OR ... OR key == value[n] |
+| `AND`    | []operation|  operation[0] AND operation[1] AND ... AND operation[n] |
+| `OR`      | []operation|  operation[0] OR operation[1] OR ... OR operation[n] |
+
+The `sort` is an optional section is an ordered array of `key:order` pairs, where `key` is a key in the state store, and the `order` is an optional string indicating sorting order: `"ASC"` for ascending and `"DESC"` for descending. If omitted, ascending order is assumed.
+
+The `pagination` is an optional section containing `limit` and `token` parameters. `limit` sets the page size. `token` is an iteration token returned by the component, and used in consecutive queries.
+
+Here is an example of the query request JSON body:
+```json
+{
+    "query": {
+        "filter": {
+            "OR": [
+                {
+                    "EQ": {"person.org": "Dev Ops"}
+                },
+                {
+                    "AND": [
+                        {
+                            "EQ": {"person.org": "Finance"}
+                        },
+                        {
+                            "IN": {"state": ["CA", "WA"]}
+                        }
+                    ]
+                }
+            ]
+        },
+        "sort": [
+            {
+               "key": "state",
+                "order": "DESC"
+            },
+            {
+                "key": "person.name"
+            }
+        ],
+        "pagination": {
+            "limit": 3
+        }
+    }
+}
+```
+
+The query request will be eventually translated into a native query language and executed in the state store component.
+
+In the case of SQL, the aforementioned query will be translated into
+```sql
+SELECT * FROM c WHERE
+  person.org = "Dev Ops" OR
+  (person.org = "Finance" AND state IN ("CA", "WA"))
+ORDER BY
+  state DESC,
+  person.name ASC
+LIMIT 3
+```
+
+Upon successful execution, the component returns JSON object with a list of matching items and the token:
+```json
+{
+  "results": [
+    {
+      "key": "1",
+      "data": {
+        "person": {
+          "code": 1001,
+          "name": "Peter",
+          "org": "Dev Ops"
+        },
+        "city": "Seattle",
+        "state": "WA"
+      },
+      "etag": "d58da615-2003-44ad-b349-cbabd54dcbfe"
+    },
+    {
+      "key": "3",
+      "data": {
+        "person": {
+          "name": "Nataly",
+          "org": "Finance",
+          "code": 1003
+        },
+        "city": "Sacramento",
+        "state": "CA"
+      },
+      "etag": "704217a7-edc9-4b5c-b027-8468b84106a9"
+    },
+    {
+      "key": "4",
+      "data": {
+        "city": "Spokane",
+        "state": "WA",
+        "person": {
+          "org": "Dev Ops",
+          "code": 1004,
+          "name": "John"
+        }
+      },
+      "etag": "256c951a-b63d-4c6c-a383-d2dc8f9d5071"
+    }
+  ],
+  "token": "5"
+}
+```
+
+## Developer guide
```

Comment:
This is a State Store Component developer guide. I would move this whole section to here https://github.com/dapr/components-contrib/tree/master/state and make it clear that you can now implement query support into state stores pointing to mongodb https://github.com/dapr/components-contrib/blob/master/state/mongodb/mongodb.go as the example. This component developer guide does not need to live here

When updating https://github.com/dapr/components-contrib/tree/master/state make it clear that this is another API that can be implemented